src/query-builder/InsertQueryBuilder.ts

```diff
@@ -335,6 +335,19 @@ export class InsertQueryBuilder<Entity> extends QueryBuilder<Entity> {
             query += ` RETURNING ${returningExpression}`;
         }
 
+
+        // Inserting a specific value for an auto-increment primary key in mssql requires enabling IDENTITY_INSERT
+        // IDENTITY_INSERT can only be enabled for tables where there is an IDENTITY column and only if there is a value to be inserted (i.e. supplying DEFAULT is prohibited if IDENTITY_INSERT is enabled)
+        if (this.connection.driver instanceof SqlServerDriver
+            && this.expressionMap.mainAlias!.hasMetadata
+            && this.expressionMap.mainAlias!.metadata.columns
+                .filter((column) => this.expressionMap.insertColumns.length > 0 ? this.expressionMap.insertColumns.indexOf(column.propertyPath) !== -1 : column.isInsert)
+                .some((column) => column.isPrimary && column.isGenerated && column.generationStrategy === "increment"
+                    && this.getValueSets().some((valueSet) => column.getEntityValue(valueSet) !== undefined && column.getEntityValue(valueSet) !== null))
+        ) {
```

Comment:
The `hasMetadata` check was required since there was a code path that led to this line without the metadata being set (rerunning the tests without that check should show that).
I've refactored that last line into a separate function, which we can use in `getInsertedColumns` as well. For an explanation, please see the comment above the if. Please feel free to let me know if there is something else you need to know.